packages/office-ui-fabric-react/src/components/TextField/TextField.tsx

```diff
@@ -122,7 +122,9 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
   public componentWillReceiveProps(newProps: ITextFieldProps): void {
     const { onBeforeChange } = this.props;
 
-    if (newProps.value !== this.state.value) {
+    // If old value prop was undefined, then component is controlled and we should
```

Comment:
component is *uncontrolled* if value is undefined, right?

Reply:
Shouldn't this just be `this.props.value !== undefined && newProps.value !== this.state.value` ?

Reply:
What if the prop previously had a value? That is the substance of #5146. If undefined is not valid for clearing state when value prop was previously defined, then we're going to need a more clear definition for behavior.

Reply:
gotcha!

Reply:
An alternative is to force controlled behavior for clearing, like passing in ``` `` ```, but I worry that `undefined` is just too close to default behavior in React to ignore even in controlled scenarios...